apps/client/src/app/pages/db/fish/fish-baits/fish-baits.component.ts

```diff
@@ -0,0 +1,40 @@
+import { ChangeDetectionStrategy, Component } from '@angular/core';
+import { LocalizedLazyDataService } from 'apps/client/src/app/core/data/localized-lazy-data.service';
+import { I18nToolsService } from 'apps/client/src/app/core/tools/i18n-tools.service';
+import { SettingsService } from 'apps/client/src/app/modules/settings/settings.service';
+import { mapValues } from 'lodash';
+import { forkJoin, of } from 'rxjs';
+import { map, shareReplay, switchMap, tap, take } from 'rxjs/operators';
+import { FishContextService } from '../../service/fish-context.service';
+
+@Component({
+  selector: 'app-fish-baits',
+  templateUrl: './fish-baits.component.html',
+  styleUrls: ['./fish-baits.component.less', '../../common-db.less'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class FishBaitsComponent {
+  public readonly spotIdFilter$ = this.fishCtx.spotId$.pipe(map((spotId) => spotId ?? -1));
+
+  public readonly loading$ = this.fishCtx.hoursByFish$.pipe(map((res) => res.loading));
+
+  public readonly baitsChartData$ = this.fishCtx.baitsByFish$.pipe(
+    switchMap((res) => {
+      if (!res.data) return of(undefined);
```

Comment:
You can return an empty array here so you don't need to use `|| []` inside the template.
```suggestion
      if (!res.data) return of([]);
```